File: compiler/testData/loadJava/kotlinSignature/propagation/parameter/SubclassFromGenericAndNot.java
package test;

import org.jetbrains.annotations.NotNull;
import jet.runtime.typeinfo.KotlinSignature;

// Extracted from KT-3302, see Kt3302 test, as well
public interface SubclassFromGenericAndNot {

    public interface NonGeneric {
        void foo(@NotNull String s);
    }

    public interface Generic<T> {
        @KotlinSignature("fun foo(key : T)")
        public void foo(T key);
    }

    public interface Sub extends NonGeneric, Generic<String> {
        @Override
        public void foo(String key);
    }
}

File: compiler/testData/loadJava/compiledKotlin/fromLoadJava/kotlinSignature/propagation/parameter/SubclassFromGenericAndNot.kt
package test

// Extracted from KT-3302, see Kt3302 test, as well
public trait SubclassFromGenericAndNot {

    public trait NonGeneric  {
        public fun foo(p0: String)

        public fun dummy() // to avoid loading as SAM interface
    }

    public trait Generic<T>  {
        public fun foo(key: T)

        public fun dummy() // to avoid loading as SAM interface
    }

    public trait Sub : NonGeneric, Generic<String> {
        override fun foo(key: String)
    }
}
